canary/scripts/test_hmr.py

```diff
@@ -0,0 +1,92 @@
+import sys
+from pathlib import Path
+
+import numpy as np
+from openforcefield.topology import Molecule
+from openmmforcefields.generators import SystemGenerator
+from pkg_resources import resource_filename
+from simtk import openmm, unit
+from simtk.openmm import app
+
+DATA_PATH = Path(resource_filename("openforcefields", "../canary/data/")).resolve()
+
+coverage_mols = DATA_PATH / "coverage.smi"
+propyne_mols = DATA_PATH / "propynes.smi"
+
+
+class NANEnergyError(Exception):
+    """Base exception for a system with NaN-like potential energy"""
+
+
+class CanaryError(Exception):
+    """Base exception for canary"""
+
+
+class HMRCanaryError(CanaryError):
+    """Exception for an HMR canary test failing"""
+
+
+def hmr_driver(mol, ff_name):
+    """Given an OpenFF Molecule, run a short 4 fs HMR simulation. This function is adapted from
+    https://github.com/openforcefield/openforcefields/issues/19#issuecomment-689816995"""
+    print(
+        f"Running HMR with force field {ff_name} and molecule with SMILES {mol.to_smiles()}"
+    )
+
+    forcefield_kwargs = {
+        "constraints": app.HBonds,
+        "rigidWater": True,
+        "removeCMMotion": False,
+        "hydrogenMass": 4
+        * unit.amu,  # Does this also _subtract_ mass from heavy atoms?:w
+    }
+
+    system_generator = SystemGenerator(
+        small_molecule_forcefield=ff_name,
+        forcefield_kwargs=forcefield_kwargs,
+        molecules=mol,
+    )
+    system = system_generator.create_system(mol.to_topology().to_openmm())
+
+    temperature = 300 * unit.kelvin
+    collision_rate = 1.0 / unit.picoseconds
+    timestep = 4.0 * unit.femtoseconds
+
+    integrator = openmm.LangevinIntegrator(temperature, collision_rate, timestep)
+    context = openmm.Context(system, integrator)
+    mol.generate_conformers(n_conformers=1)
+    context.setPositions(mol.conformers[0])
+
+    # Run for 10 ps
+    integrator.step(500)
```

Comment:
This is only 2ps

Reply:
Also, here's where we could put the `try`/`except` to catch `NaN`s from the simulation.